server/json-rpc/src/lib.rs

```diff
@@ -96,29 +99,59 @@ impl<T: SubgraphProvider> JsonRpcServer<T> {
         Ok(())
     }
 
+    fn check_subgraph_auth(
+        &self,
+        subgraph_name: String,
+        bearer_token: String,
+    ) -> Result<(), jsonrpc_core::Error> {
+        if !should_check_auth() {
+            return Ok(());
+        }
+
+        let is_authd = self
+            .store
+            .check_subgraph_name_access_token(subgraph_name, bearer_token)
+            .map_err(move |e| {
+                error!(self.logger, "Failed to check access token: {}", e);
+                json_rpc_error(JSON_RPC_INTERNAL_ERROR, "database error".to_owned())
+            })?;
+
+        if is_authd {
+            return Ok(());
+        }
+
+        return Err(json_rpc_error(
+            JSON_RPC_UNAUTHORIZED_ERROR,
+            "API key is invalid".to_owned(),
```

Comment:
This should probably be `Invalid access token`, since `API key` is not what we call this any more.

Reply:
makes sense!